activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -1225,12 +1225,16 @@ def check_constraints(table_name)
       # The +options+ hash can include the following keys:
       # [<tt>:name</tt>]
       #   The constraint name. Defaults to <tt>chk_rails_<identifier></tt>.
+      # [<tt>:if_not_exists</tt>]
+      #   (PostgreSQL only) Silently ignore if the constraint already exists, rather than raise an error.
```

Comment:
This is not PostgreSQL only, because we use it like `return if if_not_exists && check_constraint_exists?(table_name, **options)`, not special PostgreSQL syntax?

Reply:
Heh, you're right. I was thinking it was Postgres only at first and was going to do an actual `IF NOT EXISTS` in the SQL like other parts of Postgres do. But that doesn't work. Only on the `ALTER TABLE ... DROP CONSTRAINT IF EXISTS ...` side. Fixed.